.github/workflows/ci-2.x.yml

```diff
@@ -11,7 +11,33 @@ on:
       - master
 
 jobs:
-  tests:
+
+  code_style_composer_checks:
+    name: Runs PHP-CS-Fixer to check for CS issues, and validates composer.json
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.0
+          extensions: composer:v2
+
+      - name: Install PHP-CS-Fixer
+        run: composer install --no-progress --ansi --working-dir=tools/php-cs-fixer
+
+      - name: Run PHP-CS-Fixer
+        run: |
+          sed -i "s/'phpdoc_to_return_type' => true/'phpdoc_to_return_type' => false/g" .php-cs-fixer.php
+          composer cs-check
+
+      - name: Validate composer
+        run: composer validate --strict --no-check-lock
+
+  tests_4_5:
```

Comment:
you could add a dependency with previous job to prevent this one to trigger if parent job has failed

```yaml
needs: [ code_style_composer_checks ]
```